File: _test_login.py
import asyncio, sys
from playwright.async_api import async_playwright
sys.stdout.reconfigure(encoding='utf-8')

BASE = "http://localhost:3000"

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page(viewport={"width": 1280, "height": 900})
        errors = []
        page.on("console", lambda msg: errors.append(msg.text) if msg.type == "error" else None)

        # Test customer login
        await page.goto(f"{BASE}/login", timeout=15000)
        await page.wait_for_timeout(2000)
        await page.fill('input[placeholder="Enter username"]', "customer")
        await page.fill('input[placeholder="Enter your password"]', "Test123!@")
        await page.click('button[type="submit"]')
        try:
            await page.wait_for_url("**/dashboard**", timeout=15000)
            print("PASS: Customer login works")
            await page.screenshot(path="C:\\Users\\hp\\Desktop\\springwellbk\\debug_customer.png")
        except:
            url = page.url
            text = await page.inner_text("body")
            print(f"FAIL: Customer login - URL: {url}")
            print(f"  Body: {text[:300]}")
            await page.screenshot(path="C:\\Users\\hp\\Desktop\\springwellbk\\debug_customer_fail.png")

        # Test admin login
        await page.goto(f"{BASE}/login", timeout=15000)
        await page.wait_for_timeout(2000)
        await page.fill('input[placeholder="Enter username"]', "admin")
        await page.fill('input[placeholder="Enter your password"]', "Admin123!@")
        await page.click('button[type="submit"]')
        try:
            await page.wait_for_url("**/admin**", timeout=15000)
            print("PASS: Admin login works")
            await page.screenshot(path="C:\\Users\\hp\\Desktop\\springwellbk\\debug_admin.png")
        except:
            url = page.url
            text = await page.inner_text("body")
            print(f"FAIL: Admin login - URL: {url}")
            print(f"  Body: {text[:300]}")
            await page.screenshot(path="C:\\Users\\hp\\Desktop\\springwellbk\\debug_admin_fail.png")

        if errors:
            print(f"\n{len(errors)} console errors:")
            for e in errors[:5]:
                print(f"  {e[:120]}")

        await browser.close()

asyncio.run(main())

File: test-login.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage({ viewport: { width: 1280, height: 900 } });

  const consoleErrors = [];
  page.on('console', msg => { if (msg.type() === 'error') consoleErrors.push(msg.text()); });
  page.on('pageerror', err => consoleErrors.push(err.message));

  await page.goto('http://localhost:3000/login');
  await page.waitForLoadState('networkidle');
  await page.waitForTimeout(2000);

  // Check login form exists
  const form = await page.locator('form').count();
  console.log('Forms found:', form);

  await page.fill('#username', 'customer');
  await page.fill('#password', 'Test123!@');
  await page.click('button[type="submit"]');
  await page.waitForTimeout(5000);
  await page.waitForLoadState('networkidle');

  console.log('URL after submit:', page.url());
  await page.screenshot({ path: '/tmp/test-login-result.png', fullPage: true });

  if (consoleErrors.length > 0) {
    console.log('Console errors:');
    consoleErrors.forEach(e => console.log('  -', e.substring(0, 200)));
  }

  // Check for error messages on page
  const alertText = await page.locator('[role="alert"], .text-red-500, .bg-red-50').allTextContents();
  console.log('Alert/error text:', alertText);

  // If we got to dashboard, check transactions
  if (page.url().includes('dashboard')) {
    console.log('SUCCESS: On dashboard!');
    const txDivs = await page.locator('[class*="cursor-pointer"]').allTextContents();
    console.log('Clickable elements:', txDivs.length);
    if (txDivs.length > 0) {
      console.log('First clickable text:', txDivs[0].substring(0, 100));
    }
  } else {
    console.log('FAIL: Not on dashboard');
  }

  await browser.close();
})();
